content/recipes/infrastructure/gupri.md

```diff
@@ -0,0 +1,157 @@
+
+Most likely, you don’t want to set up a specialized HTTP web server at home. This would in principle be an option, but introduces enormous debt on your side: you have to guarantee it is running all the time, and secure it against malicious intents (finally, you have to pay the bill for electricity and internet access as well). You have to rent a domain name. You have to be quite an expert to set the system up and keep it running. 
+
+So, what are your options? A public service for realizing GUPRIs sounds like it would be a good idea!
+
+In the following recipe you will learn about w3id.org, purl.org, and the DOI / handle system, all of which are (or can be accessed as) free-of-cost services. 
+
+
+### Purl.org
+
+Although being the oldest and supposedly easiest-to-use service, purl.org is known by some veterans for having been offline for a long time. This was especially bad as a "persistent" identifier should be resolvable at all times. This service is run by the non-profit [Internet Archive](archive.org).
+
+To access the graphical user interface / web frontend, you need to register with archive.org. The frontend is easy to work with, although it is known for sometimes throwing errors or reacting in unexpected ways. 
+
+Using purl.org, you can add custom redirects, with [HTTP status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) 301, 302, 303, 307, or errors encoded as HTTP status codes 404 and 410. It also allows "partial redirects", i.e. it would attach all the string behind a known part to redirect you (example: deposited URL for the partial redirect is http://example.com/one and intended to be routed from http://purl.org/example . When you call http://purl.org/example/two it will route you to http://example.com/one/two . If there was no partial redirect in place, you would receive an error when calling http://purl.org/example/two because this URL would not be known to the purl.org server).
+
+
+A typical implementation route goes like this: 
+1.	Register on archive.org
+2.	Log-in to purl.org
+3.	Create your domain, e.g. purl.org/faircookbook
+4.	Create your individual redirect, e.g. purl.org/faircookbook/recipe1 with status 302 redirecting to http://example.org/faircookbook/recipe/1
+
+
+Conclusion: Quick and easy solution; but not the most reliable. You have to register one purl "domain" to one user; if multiple users want to manage a common GUPRI space, they have to share the same credentials.
+
+
+### w3id.org 
+
+The service w3id.org emerged when purl.org was down. It is run by multiple stakeholders (see self-description on w3id.org) bound by a "social contract". 
+
+Management is more difficult than purl.org, because w3id.org does not have a graphical user interface / frontend, but relies on "raw" .htaccess files deposited on github.com. You need to have some knowledge about Apache .htaccess redirect rules, thus. A clear advantage is that it gives you more freedom in the implementation.
+
```

Comment:
I would add a note about the flags that can be used: https://httpd.apache.org/docs/current/en/rewrite/flags.html